StitchCore/StitchCore/Source/Core/StitchHTTPClient.swift

```diff
@@ -207,19 +207,33 @@ internal class StitchHTTPClient {
         var isAuthenticatedRequest: Bool = true
         var refreshOnFailure: Bool = false
         var useRefreshToken: Bool = false
-        var parameters: Document = [:]
+        var data: Data?
+        var headers: [String: String]?
 
         let builder: RequestBuilder
         init(builder: @escaping RequestBuilder) throws {
             self.builder = builder
             try builder(&self)
         }
+
+        mutating func encode(withJson json: [String: Any]) throws {
+            self.data = try JSONSerialization.data(withJSONObject: json)
+        }
+
+        mutating func encode<T>(withData data: T) throws where T: Encodable {
+            if let data = data as? [String: Any] {
+                self.data = try JSONEncoder().encode(Document.init(extendedJson: data))
+            } else {
+                self.data = try JSONEncoder().encode(data)
+            }
+        }
```

Comment:
Encoding methods were added so that data could be passed to the http request body more seamlessly. Previously, you had to use a `Document`, which disallows other body types. Now you can pass in raw`Data`.